content/blog/2022/2022-02-15-getting-started-with-spring-webflux.md

```diff
@@ -0,0 +1,566 @@
+* **Elastic**: Reactive systems must be adaptive to shard or replicate components based upon their requirement. They should use predictive scaling to anticipate sudden ups and downs in their infrastructure.
+* **Message-driven**: Since all the components in a reactive system are supposed to be loosely-coupled, they must communicate across their boundaries by asynchronously exchanging of messages.
+
+{{% image alt="Reactive Straits" src="https://www.reactivemanifesto.org/images/reactive-traits.svg" %}}
+
+{{% github "https://github.com/thombergs/code-examples/tree/master/spring-webflux" %}}
+
+## Introducing Reactive Programming Paradigm
+
+Reactive programming is a programming paradigm that helps in building non-blocking, asynchronous and event-driven or message-driven perception to the world of data processing. It involves wrangling of data and events as streams that it can observe and react to the changes by processing or transformation. So we must understand the basic differences between blocking and non-blocking asynchronous request processing.
+
+### Blocking Request
+
+In a conventional MVC application, whenever a request reaches the server, a servlet thread is being created and delegated to worker threads to perform various operations like I/O, database processing, etc. While the worker threads are busy to complete its process, the servlet threads get into waiting state due to which the calls remain blocked. This is also often termed as synchronous request processing.
+
+{{% image alt="Blocking request" src="images/posts/spring-webflux/blocking_request.png" %}}
+
+### Non-blocking Request
+
+In a non-blocking system, all the incoming requests are accompanied with an event handler and a callback information. The request thread delegates the incoming request to a thread pool which manages a pretty small number of threads. Then the thread pool delegates the request to its handler function and gets available to process next incoming requests from the request thread. When the handler function completes its process, one of the thread from the pool fetches the response and pass it to the callback function. Thus the threads in a non-blocking system never goes into the waiting state. This increases the productivity and the performance of the application.
+
+{{% image alt="Non-blocking request" src="images/posts/spring-webflux/non-blocking_request.png" %}}
+
+### Reactive Libraries used in Java
+
+The landscape of reactive world in Java is quite evolving and matured enough. Every now and then we hear a new Reactive framework being introduced and consumed by different companies. Some of them include:
+
+* **RxJava**: It is implemented out of the [ReactorX](https://reactivex.io/) project which hosts implementations for multiple programming languages and platforms. *ReactiveX* is a combination of the best ideas from the *Observer* pattern, the *Iterator* pattern, and *functional programming*.
```

Comment:
This section talks about all the Reactive libraries available in Java. Since the initial portion was to focus on Reactive Programming concepts. So thought of including all theis options before someone would opt for Webflux. 

Webflux is internally built using the core components of RxJava and RxNetty.

Reply:
WebFlux is not mentioned at all in this section, so the reader will wonder why you are listing all these libraries and not mention WebFlux. Is WebFlux an alternative to them? Is it part of one of the libraries? Please either include WebFlux here or remove the whole section.

Reply:
I have added few lines to show the importance of this section. Please have a look if its fine.